### PR TITLE
chore: open join spilling by default

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

- Part of #14732 

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14843)

### CODE REVIEW

**Comment by xudong963** [comment]
Let's make join spilling open by default!

**Review by Dousir9 — APPROVED** [review]
Great, LGTM !

File: src/query/service/src/pipelines/processors/transforms/hash_join/build_spill/transform_build_spill.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ use databend_common_expression::DataBlock;
 use databend_common_sql::plans::JoinType;
 use log::info;
 
+use crate::pipelines::processors::transforms::hash_join::transform_hash_join_build::HashJoinBuildStep;
```
Comment (BohuTANG) [inline src/query/service/src/pipelines/processors/transforms/hash_join/build_spill/transform_build_spill.rs:22]:
```suggestion
use crate::pipelines::processors::transforms::hash_join::HashJoinBuildStep;
```

**Review by zhang2014 — APPROVED** [review]

File: src/query/service/src/pipelines/processors/transforms/hash_join/build_spill/transform_build_spill.rs (hunk 1 above)
Comment (BohuTANG) [reply src/query/service/src/pipelines/processors/transforms/hash_join/build_spill/transform_build_spill.rs]: We should export the `HashJoinBuildStep` to mod.rs first.

**Pull request merged**